Update laboratory1.sql

ValeriyWorld opened the pull request.

ValeriyWorld committed:
Update laboratory1.sql

UsenkoArtem commented on km51/Mironchenko_Valerii/laboratory1.sql:
```diff
@@ -1,2 +1,61 @@
+ALTER TABLE  SONG_NAME
+  ADD CONSTRAINT song_fk FOREIGN KEY (human_name_fk) REFERENCES SONG (song_name);
+  
+  
```
Забыл вставить 3 задание
/* —------------------------------------------------------------------------- 
3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
—-------------------------------------------------------------------------*/
—Код відповідь:
```
GRANT INSERT ANY TABLE valerii1
GRANT SELECT ANY TABLE valerii1
```

/*---------------------------------------------------------------------------
3.a. 
Скільки проданого найдорожчого товару?
Виконати завдання в SQL. 
4 бали
—-------------------------------------------------------------------------*/
SQL

```
Select SUM(OrderItems.quantity) 
 FROM OrderItems
  Where OrderItems.item_price = (
  Select Max(OrderItems.item_price)
   FROM OrderItems
  );

```
Алгебра кодда

```
A = (PROJECT (OrderItems) {MAX(item_price)};

Answer = PROJECT (OrderItems WHERE (OrderItems.item_price = A ))) {SUM(quantity)}

```

—Код відповідь:


/*---------------------------------------------------------------------------
3.b. 
Який PROD_ID товару, з найкоротшою назвою?
Виконати завдання в SQL. 
4 бали
—-------------------------------------------------------------------------*/

—Код відповідь:

SQL
```
Select Products.prod_id    
 FROM Products
  Where Length(Trim(Products.prod_name)) = (
  Select Min(Length(Trim(Products.prod_name)))
   FROM Products
  );
```
Алгебра кодда 
```

A = PROJECT (Products ) {MIN(LENGTH(TRIM(prod_name)))}

B = PROJECT (Products WHERE (Length(Trim(prod_name)) = A ))) {prod_id}
```


/*---------------------------------------------------------------------------
c. 
Вивести імена постачальників у верхньому регістрі,назвавши це поле vendor_name, що не мають жодного товару.
Виконати завдання в алгебрі Кодда. 
4 бали

SQL:
```
SELECT Upper(Vendors.vend_name) as "vendor_name"
FROM Vendors
MINUS
SELECT Upper(Vendors.vend_name)
FROM Vendors, Products
Where Vendors.vend_id = Products.vend_id;
```
Алгебра Кодда

```
A =  Project (Vendors Times Products Where Vendors.vend_id = Products.vend_id) {vend_name}

B = Project (Vendors) {vend_name}

Answer = Project (B MINUS A) {UPPER(RENAME(vend_name, "vendor_name"))}
```

ValeriyWorld replied:
Добавлено условие задания №3(и подзаданий) с их решениями.

UsenkoArtem commented on km51/Mironchenko_Valerii/laboratory1.sql:
```diff
@@ -1,2 +1,61 @@
+GRANT "CONNECT" TO valerii;
+
+GRANT CREATE ANY TABLE TO valerii;
+
```
В задание написано создать схему с фамилией, а тут создано по имени нужно поменять что бы соответствовало поставленному заданию.

ValeriyWorld replied:
Исправлено.

UsenkoArtem commented on km51/Mironchenko_Valerii/laboratory1.sql:
```diff
@@ -1,2 +1,61 @@
+  
+ALTER TABLE  SONG_NAME
+  ADD CONSTRAINT song_fk FOREIGN KEY (human_name_fk) REFERENCES SONG (song_name);
+  
```
Нужно создать три таблицы 

- **Первая таблица (Singer)**  
Будет отвечать за певца. 
В ней будет такие поля:
**singer_id number(10)** - отвечает за уникальность певца.
**singer_first_name varchar(40)** - имя певца.
**singer_last_name  varchar(40)** - фамилия певца.
**signer_age number(3)** - возраст певца.
**singer_genre varchar(40)** - жанр в котором поет певец. 
Каждое поле должно проверяться с помощью регулярного выражения:
  - **singer_id** ^[0-9]+$
  - **singer_first_name** ^[A-Z]{1}[a-z]{0,39}$ 
  - **singer_last_name** ^[A-Z]{1}[a-z]{0,39}$
  - **singer_age** ^[1-9]{1}[0-9]{0,2}$
  - **singer_genre** ^[A-Z]{1}[a-z]{0,39}$

- **Вторая таблица (Song)**  
Будет отвечать за песню. 
В ней будет такие поля:
**song_id number(10)** - отвечает за уникальность певца.
**song_name varchar(40)** - имя песни.
**song_duration  varchar(40)** - длительность песни в минутах.
**song_temp varchar(40)** - темп песни.
**song_genre varchar(40)** - жанр в песни. 
Каждое поле должно проверяться с помощью регулярного выражения:
  - **song_id** ^[0-9]+$
  - **song_name** ^[A-Z]{1}[a-z]{0,39}$ 
  - **song_duration** ^[0-9]{1,3}[:][0-6]{1}[0-9]{1}$
  - **song_temp** ^[A-Z]{1}[a-z]{0,39}$
  - **song_genre** ^[A-Z]{1}[a-z]{0,39}$

- **Третья таблица (Singer_Song)**  
Будет отвечать за связь между певцом и песней. 
В ней будет такие поля:
**song_singer_id number(10)** - отвечает за уникальность поля этой таблицы.
**singer_id  number(10)** - внешний ключ который связывает эту таблицу и таблицу singer.
**song_id number(10)** - внешний ключ который связывает эту таблицу и таблицу song.
Каждое поле должно проверяться с помощью регулярного выражения:
  - **song_singer_id** ^[0-9]+$
  - **song_id**  ^[0-9]+$
  - **song_id**  ^[0-9]+$

UsenkoArtem replied:
В этой таблице нету поля human_name ,action_name есть только song_name_fk
в таблице не хватает полей которые будут ссылаться на две предыдущие таблицы

UsenkoArtem commented on km51/Mironchenko_Valerii/laboratory1.sql:
```diff
@@ -1,2 +1,61 @@
+  action_name VARCHAR2(15) NOT NULL
+);
+ALTER TABLE  ACTION
+  ADD CONSTRAINT action_prk PRIMARY KEY (human_name);  
```
В этой таблице нету поля human_name есть action_name

UsenkoArtem commented on km51/Mironchenko_Valerii/laboratory1.sql:
```diff
@@ -1,2 +1,61 @@
+  ADD CONSTRAINT song_fk FOREIGN KEY (human_name_fk) REFERENCES HUMAN (human_name);
+  
+ALTER TABLE  SONG_NAME
+  ADD CONSTRAINT song_fk FOREIGN KEY (human_name_fk) REFERENCES SONG (song_name);
```
В этой таблице нету поля human_name ,action_name есть только song_name_fk
в таблице не хватает полей которые будут ссылаться на две предыдущие таблицы

UsenkoArtem reviewed (commented):
Провел ревью

ValeriyWorld committed:
Update laboratory2.sql

ValeriyWorld committed:
Update laboratory1.sql

Исправлено название схемы данных, удовлетворяющее требованиям задания.
Добавлено задание №3 с условиями и решениями.

ValeriyWorld committed:
ERD upload

ValeriyWorld committed:
Update laboratory1.sql

Обновлено задание №2:
Переделаны таблицы. Все таблицы заполнены атрибутами, присвоены внутренние и внешние ключи.
Код создания таблиц был сгенерирован после составления "Physical diagram" в середе разработки - PowerDesigner.

ValeriyWorld committed:
Update laboratory3.sql

ValeriyWorld committed:
Add files via upload

ValeriyWorld committed:
Add files via upload

ValeriyWorld committed:
Update laboratory5.sql

ValeriyWorld committed:
Update laboratory4.sql
